lib/rules/no-import-of-scoped-modules-during-ssr.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+const formFactorModule = '@salesforce/client/formFactor';
+const featureFlagModule = '@salesforce/featureFlag';
+
+const userScopedModulesSet = new Set([
+    '@salesforce/user/Id',
+    '@salesforce/user/isGuest',
+    '@salesforce/userPermission',
+    '@salesforce/customPermission',
```

Comment:
`@salesforce/userPermission` and `@salesforce/customPermission` have a third segment in the specifier. See [this doc](https://salesforce.quip.com/fgdzAvkC7UTg#temp:C:SRQ3c0a9a892bd64597b651f7725).

Reply:
Correct, for the user-scoped modules I check if an import starts with a value from Set. So, I might not need the 3rd segment. Let me know if you think otherwise.